recipes/tls-securepair-to-tlssocket/src/workflow.ts

```diff
@@ -0,0 +1,159 @@
+
+	for (const node of newExpressions) {
+		const callee = node.field('constructor');
+		if (!callee) continue;
+
+		let newConstructorName = 'TLSSocket';
+		if (callee.kind() === 'member_expression') {
+			const object = callee.field('object');
+			if (object) {
+				newConstructorName = `${object.text()}.TLSSocket`;
+			}
+		}
+
+		// Replace the constructor call with `new TLSSocket(socket)`.
+		edits.push(node.replace(`new ${newConstructorName}(socket)`));
+		const declarator = getClosest(node, ['variable_declarator']);
+		if (declarator) {
+			const idNode = declarator.field('name');
+			if (idNode) {
+				const oldName = idNode.text();
+				let newName = 'socket';
+				if (oldName !== 'pair' && oldName !== 'SecurePair') {
+					if (oldName.includes('Pair'))
+						newName = oldName.replace('Pair', 'Socket');
+					else if (oldName.includes('pair'))
+						newName = oldName.replace('pair', 'socket');
+				}
+
+				// Find usages like `pair.cleartext` or `pair.encrypted` to remove those
+				// statements as they don't apply to TLSSocket-based code.
+				const obsoleteUsages = rootNode.findAll({
+					rule: {
+						kind: 'member_expression',
+						all: [
+							{ has: { field: 'object', regex: `^${oldName}$` } },
+							{ has: { field: 'property', regex: '^(cleartext|encrypted)$' } },
+						],
+					},
+				});
+				for (const usage of obsoleteUsages) {
+					const statement = getClosest(usage, [
+						'lexical_declaration',
+						'expression_statement',
+					]);
+					if (statement) linesToRemove.push(statement.range());
+				}
+
+				// Rename the variable (e.g. `pair` -> `socket`) and update references.
+				edits.push(idNode.replace(newName));
+				const references = rootNode.findAll({
+					rule: { kind: 'identifier', regex: `^${oldName}$` },
+				});
+
+				// Update all references, skipping property accesses and imports.
+				for (const ref of references) {
+					const parent = ref.parent();
+					if (!parent) continue;
+
+					const parentKind = parent.kind();
+					if (parentKind === 'member_expression') {
+						const property = parent.field('property');
+						if (property && property.id() === ref.id()) continue;
+					}
+					if (
+						parentKind === 'import_specifier' ||
+						parentKind === 'shorthand_property_identifier_pattern'
+					)
+						continue;
+					if (ref.id() === idNode.id()) continue;
+
+					edits.push(ref.replace(newName));
+				}
+			}
+		}
+	}
+
+	const sourceCode = rootNode.commitEdits(edits);
+
+	let output = removeLines(sourceCode, linesToRemove) ?? '';
+	// Normalize newlines and trim trailing whitespace for predictable snapshots.
+	output = output.replace(/\r\n/g, '\n').replace(/\r/g, '\n');
+	output = output.replace(/(^.*\S)[ \t]+$/gm, '$1');
+	output = output.replace(/^\uFEFF/, '');
+
+	const eol =
+		typeof process !== 'undefined' && process.platform === 'win32'
+			? '\r\n'
+			: '\n';
+	output = output.replace(/\n/g, eol);
+	if (output.endsWith(eol)) output = output.slice(0, -eol.length);
```

Comment:
I don’t think it’s worth keeping this there if it’s only for Windows tests. We can find another solution for that.